src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/BundleMapper.java

```diff
@@ -30,18 +30,22 @@ public Bundle mapToBundle(final MedicalReport medicalReport) {
         final Bundle bundle = generateInitialPathologyBundle();
         final Consumer<Resource> addToBundle = resource -> addResourceToBundle(bundle, resource);
 
-        addToBundle.accept(medicalReport.getPatient());
-
-        Optional.ofNullable(medicalReport.getRequestingPractitioner()).ifPresent(addToBundle);
-        Optional.ofNullable(medicalReport.getRequestingOrganization()).ifPresent(addToBundle);
-        Optional.ofNullable(medicalReport.getPerformingPractitioner()).ifPresent(addToBundle);
-        Optional.ofNullable(medicalReport.getPerformingOrganization()).ifPresent(addToBundle);
-        Optional.ofNullable(medicalReport.getTestRequestSummary()).ifPresent(addToBundle);
-        Optional.ofNullable(medicalReport.getTestReport()).ifPresent(addToBundle);
-
-        medicalReport.getSpecimens().forEach(addToBundle);
-
-        medicalReport.getTestResults().forEach(addToBundle);
+        Stream.of(
+            Stream.of(
+                medicalReport.getMessageHeader(),
+                medicalReport.getPatient(),
+                medicalReport.getRequestingPractitioner(),
+                medicalReport.getRequestingOrganization(),
+                medicalReport.getPerformingPractitioner(),
+                medicalReport.getPerformingOrganization(),
+                medicalReport.getTestRequestSummary(),
+                medicalReport.getTestReport()
+            ),
+            medicalReport.getSpecimens().stream(),
+            medicalReport.getTestResults().stream()
+        )
+            .flatMap(i -> i)
```

Comment:
Interesting way to do it. A couple of alternatives that _might_ make it more obvious what's going on:
```suggestion
            .flatMap(Function.identity())
```
or
```suggestion
            .reduce(Stream.empty(), Stream::concat)
```